job-descriptions/software-engineer.md

```diff
@@ -33,14 +35,36 @@ We use a variety of technologies to help us accomplish our goals:
 - Backend: Go, GraphQL, PostgreSQL, Redis
 - Deployment and operations: Kubernetes, Docker, Google Cloud Platform, Terraform
 
-We expect that you have at least two years of experience relevant to the problems that you will be working on and/or the technologies that you will be using at Sourcegraph.
+The actual problems that you will work on and the technologies that you will use depend on your exact interests and skills. We'll figure out where you fit on the team as part of the interview process.
+
+#### Your responsibilities
+
+You will:
+
+- Collaborate with the product team to define and prioritize the problems that you will be working to solve.
+- Write technical design documents to communicate your implementation plan and solitic feedback from teammates.
+- Write code as necessary to achieve the goals of your projects.

```

Comment:
What were you trying to get across with 'as necessary'? It is giving me the impression that writing code is not a major part of this role, or will only be done when necessary.

```suggestion
- Write code to achieve the goals of your projects.
```

Reply:
I was trying to communicate something along the lines of "your job will involve writing code, but ideally you want to achieve your goals by writing the minimum amount of code necessary". Maybe this point is too subtle for a job description or maybe I just need to find a better way to express it.

Reply:
Yeah, too subtle here I think. You could potentially say: "Write minimum amount of code to achieve the goals of your projects", but again, I think it fails to capture the point that we expect you to be thoughtful and minimal in your solutions, rather than, we don't want you to write a lot of code, which is not true.